app/services/office_code_service.rb

```diff
@@ -1,8 +1,12 @@
 class OfficeCodeService
```

Comment:
would a version namespacing approach avoid the if/case stuff here and in `provider_data_api_client` ?

Reply:
I think you're right generally tbh, if this were a case where we are completely overhauling to a new endpoint that has new logic for everything and we need to build out v1 while running v0 for users in parallel, name-spacing would be appropriate. This case feel more like us getting a single broken api endpoint and updating it due to a fix

Reply:
If the changes don't work, we can revert the commit. I'd rather that than introducing extra branches that will rot, and yeah generally I think versioning the namespaces is a good idea, but it'll only really save hassle if we want to say use v1 of one endpoint and v2 of another.

I'd say just use the most appropriate endpoint for the method for now, and if we do hit a problem then we can add in versioned namespaces.

Reply:
the structure of the client endpoints _shouldnt_ be too volatile in terms of their structure i'd guess - id expect response payloads to evolve over time more though

Reply:
That's a point, another reason to not yet though is I don't think there are any other v1 endpoints; but I don't know the full structure of the API so will defer the decision to Ivan.

Reply:
So: 
I have not yet reviewed the other v0 to v1 changes so I don't know how big an overhaul it is. There is basically a new version of each endpoint + some new ones, but I don't know whether they've actually changed the backend/structure for all of them or not. It's probably worth us going over the new endpoints to see if there's anything there that can be used to enhance our user experience. If we do find things that lead to us deciding "ok, lets port everything to this new endpoint" I think namespacing is a good idea, removing tech debt via deleting folder that has v1 stuff in it, to me, is safer/easier than reverting a commit (what if we ended up adding more stuff to the v0 version over time?)

Reply:
Ha, okay yeah ignore my latest comment then and go for the simple approach and we can look at the wider API in the new year